cluster_pack/filesystem.py

```diff
@@ -8,6 +8,12 @@
 
 try:
     from s3fs import S3FileSystem
+
+    def _makedirs(self, path, exist_ok=False):
+        bucket, _, _ = self.split_path(path)
+        if not self.exists(bucket):
+            self.mkdir(bucket)
+    S3FileSystem.makedirs = _makedirs

```

Comment:
Why do you need to overwite S3 makedirs function if you never use it ?

Reply:
- arrow calls mkdirs https://github.com/apache/arrow/blob/master/python/pyarrow/filesystem.py#L336
- in S3FS mkdirs is an alias to makedirs which is implemented as no-op (def makedirs() pass:)

I fixed it with https://github.com/dask/s3fs/pull/331 but we need to wait for a release.

Added a comment in the code.